openmc/statepoint.py

```diff
@@ -389,7 +389,7 @@ def tallies(self):
                         new_filter.mesh = self.meshes[key]
 
                     # Add Filter to the Tally
-                    tally.add_filter(new_filter)
+                    tally.filters.append(new_filter)
```

Comment:
My issue with using `append` in something so fundamental as `StatePoint` is no error checking is performed (albeit everything *should* be fine without error checking, it has been useful for me during the early stages of development of the API to have some error checking here). What do you think about reworking this code to create a list of `Filter` and then assign that list to `tally.filters`?